addon/-private/model.ts

```diff
@@ -168,69 +180,58 @@ export default class Model extends EmberObject {
   }
 
   disconnect(): void {
-    this._store = undefined;
+    this.#store = undefined;
   }
 
-  willDestroy(): void {
+  destroy(): void {
     const cache = this.store.cache;
     if (cache) {
       cache.unload(this);
     }
   }
 
+  notifyPropertyChange(key: string) {
+    Reflect.getMetadata('orbit:notifier', this, key)(this);
+  }
+
   private get store(): Store {
-    if (!this._store) {
+    if (!this.#store) {
       throw new Error('record has been removed from Store');
     }
 
-    return this._store;
+    return this.#store;
   }
 
   static get keys(): Dict<KeyDefinition> {
-    const map: Dict<KeyDefinition> = {};
-
-    this.eachComputedProperty((name, meta) => {
-      if (meta.isKey) {
-        meta.name = name;
-        map[name] = {
-          primaryKey: meta.options.primaryKey
-        };
-      }
-    }, {});
-
-    return map;
+    return this.getPropertiesMeta('key');
   }
 
   static get attributes(): Dict<AttributeDefinition> {
-    const map: Dict<AttributeDefinition> = {};
-
-    this.eachComputedProperty((name, meta) => {
-      if (meta.isAttribute) {
-        meta.name = name;
-        map[name] = {
-          type: meta.options.type
-        };
-      }
-    }, {});
-
-    return map;
+    return this.getPropertiesMeta('attribute');
   }
 
   static get relationships(): Dict<RelationshipDefinition> {
-    const map: Dict<RelationshipDefinition> = {};
-
-    this.eachComputedProperty((name, meta) => {
-      if (meta.isRelationship) {
-        meta.name = name;
-        map[name] = {
-          kind: meta.options.kind,
-          type: meta.options.type,
-          inverse: meta.options.inverse,
-          dependent: meta.options.dependent
-        };
+    return this.getPropertiesMeta('relationship');
+  }
+
+  static getPropertiesMeta(kind: string) {
+    const properties = Object.getOwnPropertyNames(this.prototype);
+    const metas = {};
```

Comment:
```suggestion
    const meta = {};
```

I think that `meta`, like `data`, should technically be considered plural (and I'm not sure the `s` adds clarity). Sorry to be pedantic about naming!